dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```diff
@@ -229,15 +231,16 @@ private String getTimeoutMessage(boolean scan) {
 
     private static class TimeoutCheckTask implements TimerTask {
 
-        private DefaultFuture future;
+        private final Long requestID;
 
-        TimeoutCheckTask(DefaultFuture future) {
-            this.future = future;
+        TimeoutCheckTask(Long requestID) {
+            this.requestID = requestID;
         }
 
         @Override
         public void run(Timeout timeout) {
-            if (future == null || future.isDone()) {
+            DefaultFuture future = DefaultFuture.getFuture(requestID);
+            if (future.isDone()) {
```

Comment:
Seems like we should judge null for future since the normal response will have called `DefaultFuture.remove`.

Reply:
good cache, done.